### ISSUE TITLE
The problem when using intl extension with a number formatter prototype

### ISSUE THREAD

**Issue by emr** [twigphp/Twig#3140]
New intl extension is working properly but I want to give the project's number formatter for configuring global number (and currency) formats. When the passing `$numberFormatterPrototype` argument of `Twig\Extra\Intl\IntlExtension::__construct()`, I get following error:

```
The number formatter attribute "1" does not exist, known attributes are: "grouping_used", "decimal_always_shown", "max_integer_digit", "min_integer_digit", "integer_digit", "max_fraction_digit", "min_fraction_digit", "fraction_digit", "multiplier", "grouping_size", "rounding_mode", "rounding_increment", "format_width", "padding_position", "secondary_grouping_size", "significant_digits_used", "min_significant_digits_used", "max_significant_digits_used", "lenient_parse".
```
_thrown in IntlExtension.php->createNumberFormatter (line 321)_

Here is my service configuration:

```yml
NumberFormatter:
    public: true
    shared: false
    class: NumberFormatter
    arguments:
        - ~
        - !php/const \NumberFormatter::DECIMAL
    calls:
        - method: setAttribute
          arguments: [!php/const \NumberFormatter::FRACTION_DIGITS, 2]
        # and other attribute settings...

Twig\Extra\Intl\IntlExtension:
    autowire: true
```

I think the cause is attributes which get from the formatter prototype was not mapping properly.

**Comment by emr** [comment]
@stof thank you for labeling.

**Comment by xabbuh** [comment]
I'd guess that you need to fix your service definition like this:

```yaml
NumberFormatter:
    public: true
    shared: false
    class: NumberFormatter
    arguments:
        - ~
        - !php/const \NumberFormatter::DECIMAL
    calls:
        - method: setAttribute
          arguments: ['fraction_digit', 2]
```

**Comment by emr** [comment]
@xabbuh I don't think so. The first argument of `\NumberFormatter::setAttribute()` must be an integer:

```php
public NumberFormatter::setAttribute ( int $attr , int $value ) : bool
```
See: https://www.php.net/manual/en/numberformatter.setattribute.php

**Comment by fabpot** [comment]
fixed in #3198

**Comment by emr** [comment]
@fabpot thank you so much.
